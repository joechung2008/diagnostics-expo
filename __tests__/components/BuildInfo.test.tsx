import { render } from '@testing-library/react-native';
import BuildInfo from '../../components/BuildInfo';

// Mock the ThemeContext
jest.mock('../../components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      tableBorder: '#cccccc',
      tableHeaderBg: '#e0e0e0',
    },
  }),
}));

describe('BuildInfo', () => {
  it('renders build version correctly', () => {
    const { toJSON } = render(<BuildInfo buildVersion="1.0.0" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('displays the correct table structure', () => {
    const { toJSON } = render(<BuildInfo buildVersion="2.1.3" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
