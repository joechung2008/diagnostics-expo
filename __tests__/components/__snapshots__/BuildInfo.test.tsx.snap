// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildInfo displays the correct table structure 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Build Version
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        2.1.3
      </Text>
    </View>
  </View>
</View>
`;

exports[`BuildInfo renders build version correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderColor": "#cccccc",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#cccccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Build Version
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#000000",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        1.0.0
      </Text>
    </View>
  </View>
</View>
`;
